plugin/src/cli_utils/localpvzfs/volume/types.rs

```diff
@@ -0,0 +1,266 @@
+use super::Error;
+use k8s_openapi::api::core::v1::PersistentVolume;
+use k8s_openapi::NamespaceResourceScope;
+use kube::{api::ObjectMeta, Resource, ResourceExt};
+use serde::{Deserialize, Serialize};
+use std::borrow::Cow;
+
+impl Resource for ZfsVolume {
+    type DynamicType = ();
+
+    fn kind(_: &Self::DynamicType) -> Cow<'_, str> {
+        Cow::Borrowed("ZFSVolume")
+    }
+
+    fn group(_: &Self::DynamicType) -> Cow<'_, str> {
+        Cow::Borrowed("zfs.openebs.io")
+    }
+
+    fn version(_: &Self::DynamicType) -> Cow<'_, str> {
+        Cow::Borrowed("v1")
+    }
+
+    fn plural(_: &Self::DynamicType) -> Cow<'_, str> {
+        Cow::Borrowed("zfsvolumes")
+    }
+
+    fn meta(&self) -> &ObjectMeta {
+        &self.metadata
+    }
+
+    fn meta_mut(&mut self) -> &mut ObjectMeta {
+        &mut self.metadata
+    }
+
+    type Scope = NamespaceResourceScope;
+}
+
+/// ZFSVolume struct mirroring the crd.
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub(crate) struct ZfsVolume {
+    #[serde(alias = "apiVersion")]
+    api_version: String,
+    kind: String,
+    metadata: ObjectMeta,
+    spec: ZfsVolSpec,
+    status: ZfsVolStatus,
+}
+
+/// Spec for the ZFSVolume cr.
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub(crate) struct ZfsVolSpec {
+    /// Capacity of volume.
+    capacity: String,
+    /// Specifies the block-level compression algorithm to be applied.
+    compression: String,
+    /// Dedup on or off on the volume.
+    dedup: String,
+    /// Encryption on or off. This can be algorithm used also.
```

Comment:
Yes, User can set it "on" and we use any default compresssion algorithm. Otherwise, they can specify a specifc supported algorithm.

https://github.com/openebs/zfs-localpv/blob/develop/deploy/yamls/zfsvolume-crd.yaml#L107